tests/libvmi/cloudinit.go

```diff
@@ -34,9 +34,11 @@ func WithCloudInitNoCloudUserData(data string, b64Encoding bool) Option {
 		volume := getVolume(vmi, diskName)
 		if b64Encoding {
 			encodedData := base64.StdEncoding.EncodeToString([]byte(data))
+			volume.CloudInitNoCloud.UserData = ""
```

Comment:
I do not understand the use case of running `WithCloudInitNoCloudUserData` twice on the same VMI. 

Could you explain it ?

Reply:
I attempted to explain this in the [commit message](https://github.com/kubevirt/kubevirt/pull/7342/commits/a7f4db59e1d35dc919b3a28a96ebf138a12b276c). 

NewCirros has a default non-empty cloud-init value. To allow the user
override it with a specific user data such as in `libvmi.NewCirros(libvmi.WithCloudInitNoCloudUserData(userData, false))`, we need to ensure the default one is not left behind.

Reply:
Maybe I've expressed myself poorly: I'm interesting in understanding the reason why would someone want to call `WithCloudInitNoCloudUserData`  twice on the same VMI. 

I understand the fix. I just don't imagine myself ever wanting to do what the fix enables.

Reply:
I don't think that I understand your point, then.

No test should ever call `libvmi.New(WithCloudInitNoCloudUserData(1), WithCloudInitNoCloudUserData(2), ...)`; but my suggested code does this implicitly in `libvmi.NewCirros(libvmi.WithCloudInitNoCloudUserData(userData, false))`

Do you find this solution bad/confusing? Do you have a better idea to let `libvmi.NewCirros()` users modify its default user data?

Reply:
Ah. I now see what you mean. It's been too long since I used any of this; I had to look at the code to realize what "NewCirros has a default non-empty cloud-init value" meant. It's [exactly what it says](https://github.com/kubevirt/kubevirt/blob/4c72454beeb171076f73170cf661f00277d15ae2/tests/libvmi/factory.go#L84). My bad.

I actually think the correct thing to do is again to have 2 types of cirros VMIs - one with, one without user data. `NewCirros` , and `NewCirrosWithCloudInit` . I find that more explicit. 

I'm OK with letting this in and eventually discuss if we want to take that route for all VM OSs, since we're discussing the exact same thing - but for fedora - on https://github.com/kubevirt/kubevirt/pull/7342/#discussion_r824689921.